### DISCUSSION TITLE
How to get the number of rows that are filtered and available in total?

### DISCUSSION THREAD

**Discussion by xkomiks** [TanStack/table#4916] [Q&A] [answered]
I use but it looks like a hack

```
const totalCount = tableInstance.getFilteredRowModel().rows.length;
```

Is it possible to get the number of rows that is available in total, already filtered?

**Comment by KevinVandy** [comment]
Why does it feel like a hack? That's how you do it. There are multiple row models to choose from too.

**Comment by xkomiks** [comment]
I don't know if you have any plans, maybe add a helper like `tableInstance.getPageCount()` but `tableInstance.getRowCount();`

But if you think it's ok, it works for me.

**Reply by KevinVandy** [reply, marked as answer]
Behind the scenes, that API would still do the above. All of the getters in the API have some memoization logic behind the scenes if you are worried about performance
